### PR TITLE
Fix default for num-nodes in ginkgo-e2e.sh

### PR DESCRIPTION
**Pull request by pmorie**
@ihmccreery

### CODE REVIEW

**Comment by ihmccreery** [comment]
Isn't this what `config-default.sh` and `config-test.sh` are for?  I don't think a user should have to specify `NUM_NODES` in order to get a functioning cluster.

**Comment by pmorie** [comment]
@ihmccreery We have a script in openshift that calls into this script directly to run the upstream E2Es against an openshift cluster -- and it broke for me due to missing default.  Agree with you that config-default/test are for this, but nonetheless, the default as implemented in this script is wrong -- it will result in `--num-nodes= ` being passed if `NUM_NODES` env var isn't set.

**Comment by ihmccreery** [comment]
Ah, you are right.

**Comment by pmorie** [comment]
@k8s-bot test this issue: #21451

**Comment by pmorie** [comment]
@k8s-bot test this issue: #21463

**Comment by ixdy** [comment]
Why don't we set a reasonable default in `test/e2e/e2e.go` and only pass `--num-nodes` if `NUM_NODES` is set? I'd rather not have defaults in `hack/ginkgo-e2e.sh`.

**Comment by pmorie** [comment]
@ixdy There are a couple other defaults in the file, but I will change it if you want, LMK

**Comment by ihmccreery** [comment]
@pmorie I agree with @ixdy; we should try hard to consolidate defaults.  Right now `test/e2e/e2e.go` I guess is the right place to do that, though I'd hear an argument to change it.

cc @kubernetes/sig-testing

**Comment by pmorie** [comment]
@ihmccreery if think the salient point for me at this moment is that we should only apply the argument if NUM_NODES is set -- we should not apply a default that breaks you if your environment isn't set up just-so.

**Comment by pmorie** [comment]
@ihmccreery @ixdy Are you guys cool with a change to only apply `--num-nodes` if `NUM_NODES` is set?

**Comment by ixdy** [comment]
SGTM

**Comment by pmorie** [comment]
@ihmccreery @ixdy PTAL

File: hack/ginkgo-e2e.sh [hunk 1]
```diff
@@ -105,8 +109,8 @@ export PATH=$(dirname "${e2e_test}"):"${PATH}"
   --cluster-tag="${CLUSTER_ID:-}" \
   --repo-root="${KUBE_VERSION_ROOT}" \
   --node-instance-group="${NODE_INSTANCE_GROUP:-}" \
-  --num-nodes="${NUM_NODES:-}" \
   --prefix="${KUBE_GCE_INSTANCE_PREFIX:-e2e}" \
+  ${num_nodes_arg} \
```
Comment (ihmccreery) [inline hack/ginkgo-e2e.sh]: You can do this with:

```
${NUM_NODES:+"--num-nodes=${NUM_NODES}"}
```

Liked `E2E_MIN_STARTUP_PODS` below.

Is it worth just changing all of these (`prefix`, `node-instance-group`, etc.)?

**Pull request merged**